code/modules/modular_computers/hardware/card_slot.dm

```diff
@@ -76,53 +63,58 @@
 	return TRUE
 
 
-/obj/item/computer_hardware/card_slot/try_eject(slot=0, mob/living/user = null, forced = 0)
-	if(!stored_card && !stored_card2)
+/obj/item/computer_hardware/card_slot/try_eject(mob/living/user = null, forced = 0)
+	if(!stored_card)
 		to_chat(user, "<span class='warning'>There are no cards in \the [src].</span>")
 		return FALSE
 
-	var/ejected = 0
-	if(stored_card && (!slot || slot == 1))
-		if(user)
-			user.put_in_hands(stored_card)
-		else
-			stored_card.forceMove(drop_location())
-		stored_card = null
-		ejected++
-
-	if(stored_card2 && (!slot || slot == 2))
-		if(user)
-			user.put_in_hands(stored_card2)
-		else
-			stored_card2.forceMove(drop_location())
-		stored_card2 = null
-		ejected++
-
-	if(ejected)
-		if(holder)
-			if(holder.active_program)
-				holder.active_program.event_idremoved(0, slot)
-
-			for(var/I in holder.idle_threads)
-				var/datum/computer_file/program/P = I
-				P.event_idremoved(1, slot)
-		if(ishuman(user))
-			var/mob/living/carbon/human/H = user
-			H.sec_hud_set_ID()
-		to_chat(user, "<span class='notice'>You remove the card[ejected>1 ? "s" : ""] from \the [src].</span>")
-		playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, FALSE)
-		return TRUE
-	return FALSE
+	if(user)
+		user.put_in_hands(stored_card)
+	else
+		stored_card.forceMove(drop_location())
+	stored_card = null
+
+	if(holder)
+		if(holder.active_program)
+			holder.active_program.event_idremoved(0)
+
+		for(var/I in holder.idle_threads)
+			var/datum/computer_file/program/P = I
+			P.event_idremoved(1)
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		H.sec_hud_set_ID()
+	to_chat(user, "<span class='notice'>You remove the card from \the [src].</span>")
+	playsound(src, 'sound/machines/terminal_insert_disc.ogg', 50, FALSE)
+	return TRUE
 
 /obj/item/computer_hardware/card_slot/attackby(obj/item/I, mob/living/user)
 	if(..())
 		return
 	if(I.tool_behaviour == TOOL_SCREWDRIVER)
-		to_chat(user, "<span class='notice'>You press down on the manual eject button with \the [I].</span>")
-		try_eject(0,user)
-		return
+		if(stored_card)
+			to_chat(user, "<span class='notice'>You press down on the manual eject button with \the [I].</span>")
+			try_eject(0,user)
```

Comment:
```suggestion
			try_eject(user)
```
Better look for all calls of `try_eject` and make sure you remove the first argument if it still refers to `slot`

Reply:
Oof, I went and missed the one that was in the same file.

Did a search, and removed any reference to the slot number in the rest of the code. I also removed slot references in the `try_eject()` proc for the AI reader and battery module (both of which did not use slots but I assume had the arg to be consistent with the card reader).